src/operations/indexes.ts

```diff
@@ -318,38 +303,28 @@ export class EnsureIndexOperation extends CreateIndexOperation {
     this.collectionName = collectionName;
   }
 
-  override executeCallback(
-    server: Server,
-    session: ClientSession | undefined,
-    callback: Callback
-  ): void {
+  override async execute(server: Server, session: ClientSession | undefined): Promise<string> {
     const indexName = this.indexes[0].name;
     const cursor = this.db.collection(this.collectionName).listIndexes({ session });
-    cursor.toArray().then(
-      indexes => {
-        indexes = Array.isArray(indexes) ? indexes : [indexes];
-        if (indexes.some(index => index.name === indexName)) {
-          callback(undefined, indexName);
-          return;
-        }
-        super.executeCallback(server, session, callback);
-      },
-      error => {
-        if (error instanceof MongoError && error.code === MONGODB_ERROR_CODES.NamespaceNotFound) {
-          // ignore "NamespaceNotFound" errors
-          return super.executeCallback(server, session, callback);
-        }
-        return callback(error);
+    try {
+      const indexes = await cursor.toArray();
+      if (indexName && indexes.some(index => index.name === indexName)) return indexName;
+      return await super.execute(server, session);
+    } catch (error) {
+      if (error instanceof MongoError && error.code === MONGODB_ERROR_CODES.NamespaceNotFound) {
+        // ignore "NamespaceNotFound" errors
+        return super.execute(server, session);
       }
-    );
+      return error;
+    }
```

Comment:
The original error handling was only for the `toArray`, not the create call.  I'm not sure if `createIndex` will ever throw namespace not found errors, but we should revert the change anyways.

We can also flatten the logic a bit if we use a `catch` handler instead of a try-catch on the `toArray()`:


```suggestion
    const indexes = this.db.collection(this.collectionName)
		.listIndexes({ session })
		.toArray()
		.catch(error => {
			if (error is ns not found error) return []
			throw error;
		});

    if (indexName && indexes.some(index => index.name === indexName)
		return indexName;
    return await super.execute(server, session);
```